src/lib/snapper/snapshot_dbus.rb

```diff
@@ -0,0 +1,214 @@
+          log.debug("get_config for name '#{config_name}' result:#{result}")
+
+          result
+        end
+
+        def list_snapshots(config_name = nil)
+          result = dbus_object.ListSnapshots(config_name).first
+          log.debug("list_snapshots for name #{config_name} result:#{result}")
+          result
+        end
+
+        def dbus_object
+          @dbus_object ||= DBus::SystemBus.instance().
+            service("org.opensuse.Snapper").
+            object("/org/opensuse/Snapper")
+          @dbus_object.default_iface = "org.opensuse.Snapper"
+          @dbus_object.introspect()
+          @dbus_object
+        end
+
+        def create_comparison(config_name, num1, num2)
+          result = dbus_object.CreateComparison(config_name, num1, num2).first
+          log.debug("create_comparison config_name:#{config_name} num1:#{num1} num2:#{num2} "\
+                   "result: #{result}")
+
+          result
+        end
+
+
+        def delete_comparison(config_name, num1, num2)
+          result = dbus_object.DeleteComparison(config_name, num1, num2).first
+          log.debug("delete_comparison config_name:#{config_name} num1:#{num1} num2:#{num2} "\
+                   "result:#{result}")
+
+          result
+        end
+
+
+        def get_files(config_name, num1, num2)
+          result = dbus_object.GetFiles(config_name, num1, num2).first
+          log.debug("get_files config_name:#{config_name} num1:#{num1} num2:#{num2} result:#{result}")
+
+          result.map do |file|
+            { "filename" => unescape(file[0]), "status" => file[1] }
+          end
+        end
+
+        private
+
+        # Escape a String or Hash for snapperd. See snapper dbus documentation for details.
+        def escape(str)
+
+          ret = str.dup
+
+          if ret.is_a?(::String)
+            ret.force_encoding(Encoding::ASCII_8BIT)
+            ret.gsub!(/(\\|[\x80-\xff])/n) do |tmp|
+              if tmp == "\\"
+                "\\\\"
+              else
+                "\\x" + tmp[0].ord.to_s(16)
+              end
+            end
+          elsif ret.is_a?(Hash)
+            ret = ret.map { |k, v| [ escape(k), escape(v) ] }.to_h
+          elsif
+            raise RuntimeError, "cannot escape object"
+          end
+
+          return ret
+        end
+
+
+        # Unescape a String or Hash from snapperd. See snapper dbus documentation for details.
+        def unescape(str)
+
+          ret = str.dup
+
+          if ret.is_a?(::String)
+
+            ret.force_encoding(Encoding::ASCII_8BIT)
+            ret.gsub!(/\\(\\|x\h\h)/n) do |tmp|
+              if tmp == "\\\\"
+                "\\"
+              else
+                tmp[2,2].hex.chr
+              end
+            end
+
+          elsif ret.is_a?(Hash)
+
+            ret = ret.map { |k, v| [ unescape(k), unescape(v) ] }.to_h
+
+          elsif
+
+            raise RuntimeError, "cannot unescape object"
```

Comment:
please add to exception class of object, it really helps with debugging like `raise RuntimeError, "cannot unescape object #{ret.class}"`